ls1intum/Artemis
# `Programming exercises`: Change Docker image for Swift exercises

### Checklist
#### General



#### Changes affecting Programming Exercises

### Motivation and Context


The currently used Docker image for Swift exercise is not updated regularly (currently uses Swift 5.7 and SwiftLint 0.51.0).
However, there are newer versions of Swift and SwiftLint available. 
Therefore we now build our own docker image here: https://github.com/ls1intum/artemis-swift-swiftlint-docker
This allows us to easily update the image when new versions are available and we are no more dependent on a third party.

### Description

The new image is added to the _application.yml_ and the link in the documentation is updated.

### Steps for Testing

Prerequisites:
- 1 Instructor

**Testserver 1 (Bamboo):**

1. Create a new Swift programming exercise
2. Go to the build plans
3. Make sure the correct image is used in the build plan
4. Also make sure the exercise builds correctly

**Testserver 2 (Jenkins):**
1. Create a new Swift programming exercise
2. Go to the build plans
3. Make sure the correct image is used in the build plan
4. Also make sure the exercise builds correctly

### Review Progress






#### Code Review
#### Manual Tests

### Screenshots



Correct image in Bamboo:

Correct image in Jenkins:
<img width="1420" alt="Screenshot 2023-04-27 at 13 00 28" src="https://user-images.githubusercontent.com/9954674/234844519-beaca8a1-fc6b-4dde-b662-87689a760aff.png">
